src/services/EcoBridge/Connext/Connext.ts

```diff
@@ -0,0 +1,655 @@
+
+      let gasInUSD: string | undefined = undefined
+
+      if (gasPrice && fromChainId !== ChainId.POLYGON) {
+        const gasInGwei = BigNumber.from(gasPrice).mul(prepare)
+        const totalGas = formatUnits(gasInGwei)
+
+        const {
+          bundle: { nativeCurrencyPrice },
+        } = await request(subgraphClientsUris[fromChainId as SWPRSupportedChains], QUERY_NATIVE_PRICE, {
+          chainId: fromChainId,
+        })
+
+        if (nativeCurrencyPrice) {
+          gasInUSD = `${(Number(totalGas) * Number(nativeCurrencyPrice)).toFixed(2)}$`
+        }
+      }
+
+      const details = {
+        receiveAmount: Number(formatUnits(amountReceived, decimals)).toFixed(2),
+        fee,
+        estimateTime: '15 MIN',
+        requestId: helperRequestId,
+        gas: gasInUSD,
+      }
+
+      this.store.dispatch(this.actions.setBridgeDetails(details))
+    } catch (e) {
+      console.log(e)
+      this.store.dispatch(this.actions.setBridgeDetailsStatus({ status: SyncState.FAILED }))
+    }
+  }
+  public validate = async () => {
+    try {
+      if (!this._activeProvider || !this._account || !this._quote) return
+
+      this.store.dispatch(
+        ecoBridgeUIActions.setStatusButton({ label: 'Loading', isLoading: true, isError: false, isApproved: false })
+      )
+
+      const {
+        from: { address: fromTokenAddress, chainId: fromChainId },
+      } = this.store.getState().ecoBridge.ui
+
+      if (fromTokenAddress === Currency.getNative(fromChainId).symbol) {
+        this.store.dispatch(
+          ecoBridgeUIActions.setStatusButton({
+            label: 'Bridge',
+            isLoading: false,
+            isError: false,
+            isApproved: true,
+            isBalanceSufficient: true,
+          })
+        )
+        return
+      }
+
+      const token = new Contract(fromTokenAddress, ERC20.abi, this._activeProvider.getSigner())
+
+      const managerContract = getDeployedTransactionManagerContract(fromChainId)
+
+      if (!managerContract) throw new Error('Cannot get managerContractAddress')
+
+      const allowance: BigNumber = await token.allowance(this._account, managerContract.address)
+
+      const {
+        bid: { amount },
+      } = this._quote
+
+      if (allowance.gte(BigNumber.from(amount))) {
+        this.store.dispatch(
+          ecoBridgeUIActions.setStatusButton({
+            label: 'Bridge',
+            isLoading: false,
+            isError: false,
+            isApproved: true,
+            isBalanceSufficient: true,
+          })
+        )
+      } else {
+        this.store.dispatch(
+          ecoBridgeUIActions.setStatusButton({
+            label: 'Approve',
+            isLoading: false,
+            isError: false,
+            isApproved: false,
+            isBalanceSufficient: true,
+          })
+        )
+      }
+    } catch (e) {
+      this.store.dispatch(
+        ecoBridgeUIActions.setStatusButton({
+          label: 'Something went wrong',
```

Comment:
The label could be more specific

Reply:
https://github.com/levelkdev/swapr-dapp/pull/1008#discussion_r893301714

We will add general error messages